egs/zeroth_korean/s5/local/prepare_dict.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+# Copyright 2014 Vassil Panayotov
+# Apache 2.0
+
+# Modified by Lucas Jo 2017 (Altas Guide)
+# Prepare dictionary
+
+if [ $# -ne 2 ]; then
+	echo "Usage: $0 <lm-dir> <dst-dir>"
+	echo "e.g.: /data/local/lm data/local/dict_nosp"
+	exit 1
+fi
+lm_dir=$1
+dst_dir=$2
+
+mkdir -p $dst_dir || exit 1;
+
+# this file is  a copy of the lexicon we obtained from download_lm.sh process
+lexicon_raw_nosil=$dst_dir/lexicon_raw_nosil.txt
+
+if [[ ! -s "$lexicon_raw_nosil" ]]; then
+	cp $lm_dir/zeroth_lexicon $lexicon_raw_nosil || exit 1
+fi
+
+silence_phones=$dst_dir/silence_phones.txt
+optional_silence=$dst_dir/optional_silence.txt
+nonsil_phones=$dst_dir/nonsilence_phones.txt
+extra_questions=$dst_dir/extra_questions.txt
+
+echo "Preparing phone lists and clustering questions"
+(echo SIL; echo SPN;) > $silence_phones
+#( echo SIL; echo BRH; echo CGH; echo NSN ; echo SMK; echo UM; echo UHH ) > $silence_phones
+echo SIL > $optional_silence
+# nonsilence phones; on each line is a list of phones that correspond
+# really to the same base phone.
+awk '{for (i=2; i<=NF; ++i) { print $i; gsub(/[0-9]/, "", $i); print $i}}' $lexicon_raw_nosil |\
+	sort -u |\
+	perl -e 'while(<>){
+chop; m:^([^\d]+)(\d*)$: || die "Bad phone $_";
```

Comment:
please indent these a bit.  they read like bash commands if they start at the very left.

Reply:
indentation done